sdk/src/androidTest/java/ly/count/android/sdk/UtilsTests.java

```diff
@@ -155,4 +157,57 @@ public void extractValuesFromString() {
         Assert.assertEquals("", extractResult[0]);
         Assert.assertEquals("", extractResult[1]);
     }
+
+    @Test
+    public void splitIntoParams_badValues() {
+        Assert.assertTrue(Utils.splitIntoParams(null, new ModuleLog()).isEmpty());
+        Assert.assertTrue(Utils.splitIntoParams("", new ModuleLog()).isEmpty());
+        Assert.assertTrue(Utils.splitIntoParams(" ", new ModuleLog()).isEmpty());
+    }
+
+    @Test
+    public void splitIntoParams_junkValues() {
+        Assert.assertTrue(Utils.splitIntoParams("aa,bbb", new ModuleLog()).isEmpty());
+        Assert.assertTrue(Utils.splitIntoParams("aaa=", new ModuleLog()).isEmpty());
+        Assert.assertTrue(Utils.splitIntoParams("bbb&", new ModuleLog()).isEmpty());
+
+        Map<String, String> result = Utils.splitIntoParams("aaa=bbb=ccc&ddd=eee", new ModuleLog());
+        Assert.assertEquals(1, result.size());
+        Assert.assertEquals("eee", result.get("ddd"));
+    }
+
+    @Test
+    public void splitIntoParams_validValues() {
+        Map<String, String> result = Utils.splitIntoParams("aaa=bbb", new ModuleLog());
+        Assert.assertEquals(1, result.size());
+        Assert.assertEquals("bbb", result.get("aaa"));
+
+        result = Utils.splitIntoParams("aaa=bbb&ccc=ddd", new ModuleLog());
+        Assert.assertEquals(2, result.size());
+        Assert.assertEquals("bbb", result.get("aaa"));
+        Assert.assertEquals("ddd", result.get("ccc"));
+    }
+
+    @Test(expected = AssertionError.class)
+    public void combineParamsIntoRequest_badValues_assertionError() {
+        Assert.assertNull(Utils.combineParamsIntoRequest(null));
+    }
+
+    @Test
+    public void combineParamsIntoRequest_badValues() {
+        Assert.assertTrue(Utils.combineParamsIntoRequest(new HashMap<>()).isEmpty());
+    }
+
+    @Test
+    public void combineParamsIntoRequest() {
```

Comment:
to have a pattern for naming the test methods, I would suggest the naming as
combineParamsIntoRequest() -> combineParamsIntoRequest_validValues()

Reply:
yeah you are right we can go for it. I usually do not add something for the valid test cases. But it would be better to indicate this is the valid test case. Thx I will add those and the things you have mentioned